autosklearn/pipeline/components/data_preprocessing/one_hot_encoding/one_hot_encoding.py

```diff
@@ -1,44 +1,20 @@
 import numpy as np
 import scipy.sparse
 
-import autosklearn.pipeline.implementations.OneHotEncoder
+import sklearn.preprocessing
 
 from ConfigSpace.configuration_space import ConfigurationSpace
-from ConfigSpace.hyperparameters import CategoricalHyperparameter, \
-    UniformFloatHyperparameter
-from ConfigSpace.conditions import EqualsCondition
 
 from autosklearn.pipeline.components.base import AutoSklearnPreprocessingAlgorithm
 from autosklearn.pipeline.constants import *
-from autosklearn.util.common import check_for_bool, check_none
 
 
 class OneHotEncoder(AutoSklearnPreprocessingAlgorithm):
-    def __init__(self, use_minimum_fraction=True, minimum_fraction=0.01,
-                 categorical_features=None, random_state=None):
-        # TODO pay attention to the cases when a copy is made (CSR matrices)
-
-        self.use_minimum_fraction = use_minimum_fraction
-        self.minimum_fraction = minimum_fraction
-        self.categorical_features = categorical_features
+    def __init__(self, random_state=None):
+        self.random_state = random_state
 
     def _fit(self, X, y=None):
-        self.use_minimum_fraction = check_for_bool(self.use_minimum_fraction)
-        if self.use_minimum_fraction is False:
-            self.minimum_fraction = None
-        else:
-            self.minimum_fraction = float(self.minimum_fraction)
-
-        if check_none(self.categorical_features):
-            categorical_features = []
-        else:
-            categorical_features = self.categorical_features
-
-        self.preprocessor = autosklearn.pipeline.implementations.OneHotEncoder\
-            .OneHotEncoder(minimum_fraction=self.minimum_fraction,
-                           categorical_features=categorical_features,
-                           sparse=True)
-
+        self.preprocessor = sklearn.preprocessing.OneHotEncoder(sparse=True, categories='auto')
         return self.preprocessor.fit_transform(X)
 
     def fit(self, X, y=None):
```

Comment:
Do we need this function?

Reply:
I'm also going to adopt the structure of the Imputation class here. So, just fit() and transform() methods. Also, the sklearn's OneHotEncoder returns a sparse matrix (unless we overwrite the default argument), therefore I'll get rid also of this part that checks the type of X after transformation.